src/Apps/Auction/routes.tsx

```diff
@@ -22,6 +23,50 @@ export const routes: RouteConfig[] = [
     path: "/auction-faq",
     Component: AuctionFAQ,
   },
+  {
+    path: "auction/:saleID/bid(2)?/:artworkID",
+    Component: ConfirmBidRoute,
+    render: ({ Component, props }) => {
+      if (Component && props) {
+        const { artwork, me, location } = props as any
+        if (!artwork) {
+          return <ErrorPage code={404} />
+        }
+        handleRedirect(bidRedirect(artwork.saleArtwork.sale, me), location)
+        return <Component {...props} />
+      }
+    },
+    query: graphql`
+      query routes_BidQuery($saleID: String!, $artworkID: String!) {
+        artwork(id: $artworkID) {
+          ...LotInfo_artwork
+          _id
+          id
+          saleArtwork: sale_artwork(sale_id: $saleID) {
+            ...LotInfo_saleArtwork
+            ...BidForm_saleArtwork
+            _id
+            id
+            sale {
+              _id
+              id
+              name
+            }
+          }
+        }
```

Comment:
Can this stuff be moved into a `ConfirmBidRoute` `createFragmentContainer`? The main route `query` key (ideally, but no strictly) only contains references to fragment spreads related to components in the subtree.

Reply:
Yes, for the purposes of this PR the immediate goal was to just get a skeleton with some presentation set up while some other work could not be parallelized. I only began breaking things out into fragment containers as a final step up above. But yes, this will eventually be moved in as much as possible.